### PR TITLE
Ddt 860 fetch transport document by reference

### PR DESCRIPTION
**Pull request by pimg**
resolves: DDT-860

### CODE REVIEW

File: src/main/java/org/dcsa/ebl/model/transferobjects/TransportDocumentTO.java [hunk 1]
```diff
@@ -1,57 +1,67 @@
...
 
-    @Valid
-    private ServiceType serviceTypeAtOrigin;
+  @JsonFormat(shape = JsonFormat.Shape.STRING, pattern = "yyyy-MM-dd")
+  private LocalDate issueDate;
 
-    @Valid
-    private ServiceType serviceTypeAtDestination;
+  @JsonFormat(shape = JsonFormat.Shape.STRING, pattern = "yyyy-MM-dd")
+  private LocalDate shippedOnBoardDate;
```
Comment (mp-assco) [inline src/main/java/org/dcsa/ebl/model/transferobjects/TransportDocumentTO.java:35]: onboard is one word

File: src/main/java/org/dcsa/ebl/model/transferobjects/TransportDocumentTO.java [hunk 2]
```diff
@@ -1,57 +1,67 @@
...
-    private TransportPlan transportPlan;
+  @Size(max = 3, message = "Declared Value Currency has a max size of 3.")
+  private String declaredValueCurrency;
+
+  private Double declaredValue;
+
+  private Integer numberOfRiderPages;
+
+  @Valid
+  @JsonProperty("placeOfIssue")
```
Comment (HenrikHL) [inline src/main/java/org/dcsa/ebl/model/transferobjects/TransportDocumentTO.java:55]: IS this needed?

File: src/main/java/org/dcsa/ebl/model/transferobjects/TransportDocumentTO.java [hunk 3]
```diff
@@ -1,57 +1,67 @@
...
+
+  @Valid
+  @JsonInclude(JsonInclude.Include.NON_NULL)
+  private ShippingInstructionTO shippingInstruction;
+
+  @Valid
+  @JsonInclude(JsonInclude.Include.NON_NULL)
+  private List<ChargeTO> charges;
+
+  @Valid private List<CarrierClauseTO> clauses;
```
Comment (HenrikHL) [inline src/main/java/org/dcsa/ebl/model/transferobjects/TransportDocumentTO.java:66]: clauses should be renamed to "carrierClauses" - cannot do a suggest here as I expect this to be used elsewhere also ;-)

File: src/main/java/org/dcsa/ebl/model/transferobjects/TransportDocumentTO.java (hunk 1 above)
Comment (pimg) [reply src/main/java/org/dcsa/ebl/model/transferobjects/TransportDocumentTO.java:35]: The field name is aligned with the Open API specification. So won't fix until Open API spec is changed

File: src/main/java/org/dcsa/ebl/model/transferobjects/TransportDocumentTO.java (hunk 3 above)
Comment (pimg) [reply src/main/java/org/dcsa/ebl/model/transferobjects/TransportDocumentTO.java]: changed

**Review by Abdiiir — COMMENTED** [review]
Should the submodule not be pointing to the latest master?

File: src/test/java/org/dcsa/ebl/controller/TransportDocumentControllerTest.java [hunk 4]
```diff
@@ -0,0 +1,174 @@
...
+    shippingInstructionTO.setShippingInstructionID(UUID.randomUUID().toString());
+    shippingInstructionTO.setDocumentStatus(ShipmentEventTypeCode.RECE);
+    shippingInstructionTO.setPlaceOfIssueID(locationTO.getId());
+    shippingInstructionTO.setAreChargesDisplayedOnCopies(true);
+
+    ShipmentTO shipmentTO = new ShipmentTO();
+    shipmentTO.setBooking(new BookingTO());
+    shippingInstructionTO.setShipments(List.of(shipmentTO));
+
+    transportDocumentTO = new TransportDocumentTO();
```
Comment (Abdiiir) [inline src/test/java/org/dcsa/ebl/controller/TransportDocumentControllerTest.java:93]: Duplicate Initialization?

File: src/main/java/org/dcsa/ebl/model/transferobjects/TransportDocumentTO.java (hunk 2 above)
Comment (pimg) [reply src/main/java/org/dcsa/ebl/model/transferobjects/TransportDocumentTO.java]: fixed, jsonproperty is removed

File: src/test/java/org/dcsa/ebl/controller/TransportDocumentControllerTest.java (hunk 4 above)
Comment (pimg) [reply src/test/java/org/dcsa/ebl/controller/TransportDocumentControllerTest.java:92]: fixed, duplicate initialization is removed

**Review by mp-assco — APPROVED** [review]
looks good to me

**Pull request merged**